### ISSUE TITLE
🐛 Multiple root configs don't work with config-path

### ISSUE THREAD

**Issue by nezouse** [biomejs/biome#7390]
### Environment information

```block
pnpm biome rage --formatter --linter

CLI:
  Version:                      2.2.2
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.18.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/8.15.6

Biome Configuration:
  Status:                       Not set
  Path:                         unset

Workspace:
  Open Documents:               0
```

### What happened?

Simple reproduction: https://github.com/nezouse/biome-nested-root-config
1. Install deps with pnpm
1. Run pnpm check - it should work without issues
1. Run pnpm check:strict - it should throw an issue with noConsole

Discord ticket link: https://discord.com/channels/1132231889290285117/1412802027964203119

### Expected result

One command should fail, the other should exit without errors. But it looks like that even when running biome.strict.json it gets overridden by biome.json, which should not happen

The context is that i want to have strict config for CI usage (and cli in general) and more relaxed one for IDE - so it does not yell at me while im still writing the code. The relaxed one should be called just biome.json so it gets picked up by various IDEs without additional configuration.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by JunmanChoi** [comment]
i am having same issue. any suggestions?

```
CLI:
  Version:                      2.2.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.19.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.17.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false
```

**Comment by GeneZharov** [comment]
We have another use case. We are adding Biome to an existing large project. This project cannot pass all the neccessary linting rules right now as we cannot make so many fixes at once. As a result we cannot check against all these rules at the CI level. However, we want to make sure that new sources, as well as edits to legacy sources, pass all the linting rules.

So we need two different Biome configs at the root of the project: one for CI (less strict) and one for IDE and Git hooks (more strict).

**Comment by GeneZharov** [comment]
My workaround is to swap config files with a Node script

- /biome.json - default strict config with all rules enabled
- /biome-legacy.json - less strict config for CI or manual launches

```jsonc
// package.json

"scripts": {
  "lint": "node scripts/biome.js check",
  "lint:fix": "node scripts/biome.js check --write --unsafe",
  "lint:ci": "node scripts/biome.js ci --error-on-warnings",
}
```

```js
// scripts/biome.js

import * as fs from "node:fs";
import * as cp from "node:child_process";

const DEFAULT = "biome.json";
const LEGACY = "biome-legacy.json";

function swapConfigs(swap) {
  if (swap) {
    fs.renameSync(DEFAULT, DEFAULT + ".bak");
    fs.renameSync(LEGACY, DEFAULT);
  } else {
    fs.renameSync(DEFAULT, LEGACY);
    fs.renameSync(DEFAULT + ".bak", DEFAULT);
  }
}

const args = process.argv.slice(2).join(" ");
const cmd = `biome ${args}`;

swapConfigs(true);
const child = cp.spawn(cmd, { shell: true, stdio: "inherit" });
child.on("close", (code) => {
  swapConfigs(false);
  process.exit(code);
});
```
